src/main/resources/tpcds_2_4/q23b.sql

```diff
@@ -0,0 +1,50 @@
+--q23b.sql--
+
+
+ with frequent_ss_items as
+ (select substr(i_item_desc,1,30) itemdesc,i_item_sk item_sk,d_date solddate,count(*) cnt
+  from store_sales, date_dim, item
+  where ss_sold_date_sk = d_date_sk
+    and ss_item_sk = i_item_sk
+    and d_year in (2000, 2000+1, 2000+2,2000+3)
+  group by substr(i_item_desc,1,30),i_item_sk,d_date
+  having count(*) > 4),
+ max_store_sales as
+ (select max(csales) tpcds_cmax
+  from (select c_customer_sk,sum(ss_quantity*ss_sales_price) csales
+        from store_sales, customer, date_dim
+        where ss_customer_sk = c_customer_sk
+         and ss_sold_date_sk = d_date_sk
+         and d_year in (2000, 2000+1, 2000+2,2000+3)
+        group by c_customer_sk) x),
+ best_ss_customer as
+ (select c_customer_sk,sum(ss_quantity*ss_sales_price) ssales
+  from store_sales
+      ,customer
+  where ss_customer_sk = c_customer_sk
+  group by c_customer_sk
+  having sum(ss_quantity*ss_sales_price) > (50/100.0) *
```

Comment:
Depends on what you want.  The Qualification Substitution Parameters for query23 are in Appendix B in the TPC-DS spec.

Qualification Substitution Parameters for query23.tpl:
* MONTH.01 = 2
* YEAR.01 = 2000
* TOPPERCENT=50